ethereum/smart-contract/contracts/GatewayStaking.sol

```diff
@@ -0,0 +1,49 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.19;
+
+import { IGatewayStaking } from "./interfaces/IGatewayStaking.sol";
+import { ParameterizedAccessControl } from "./ParameterizedAccessControl.sol";
+import {ContextUpgradeable} from "@openzeppelin/contracts-upgradeable/utils/ContextUpgradeable.sol";
+import {Context} from "@openzeppelin/contracts/utils/Context.sol";
+import { ERC20 } from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import { ERC4626 } from "@openzeppelin/contracts/token/ERC20/extensions/ERC4626.sol";
+
+
+contract GatewayStaking is IGatewayStaking, ParameterizedAccessControl {
+   constructor(ERC20 asset_, string memory name_, string memory symbol_) ERC4626(asset_) ERC20(name_, symbol_) {
+      _superAdmins[msg.sender] = true;
+   }
+
+   function depositStake(uint256 assests) public override returns(uint256) {
+      // Deposit stake using ERC-4626 deposit method
+      require(assests > 0, "Must deposit assets to receive shares");
+      deposit(assests, msg.sender);
```

Comment:
Here is what the deposit method looks like in the OZ implementation: https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/extensions/ERC4626.sol#L171